gwvolman/lib/zenodo.py

```diff
@@ -0,0 +1,242 @@
+"""Whole Tale publisher provider for Zenodo."""
+
+import datetime
+import json
+import logging
+import requests
+import tempfile
+
+from markdown import Markdown
+from lxml.html.clean import Cleaner
+
+from .publish_provider import PublishProvider
+
+
+_ZENODO_ALLOWED_TAGS = {
+    "a",
+    "p",
+    "br",
+    "blockquote",
+    "strong",
+    "b",
+    "u",
+    "i",
+    "em",
+    "ul",
+    "ol",
+    "li",
+    "sub",
+    "sup",
+    "div",
+    "strike",
+}
+
+
+class ZenodoPublishProvider(PublishProvider):
+    """Whole Tale publisher provider for Zenodo."""
+
+    total_updates = 3
+    current_update = 0
+
+    def update_progress(self, message=None):
+        """Send update and increment update counter."""
+        self.current_update += 1
+        self.job_manager.updateProgress(
+            total=self.total_updates, current=self.current_update, message=message
+        )
+
+    @staticmethod
+    def _render_description(description):
+        """Convert Markdown to HTML accepted by Zenodo."""
+        parser = Markdown()
+        html = parser.convert(description)
+        html_cleaner = Cleaner(
+            allow_tags=_ZENODO_ALLOWED_TAGS, remove_unknown_tags=False
+        )
+        return html_cleaner.clean_html(html)
+
+    def deposit_tale(self, deposition):
+        """Given a Zenodo deposition, upload tarball with a Tale."""
+        self.update_progress(message="Uploading data to " + self.resource_server)
+
+        stream = self.gc.sendRestRequest(
+            "get",
+            "tale/{}/export".format(self.tale["_id"]),
+            parameters={"taleFormat": "bagit"},
+            stream=True,
+            jsonResp=False,
+        )
+        with tempfile.NamedTemporaryFile(delete=False, suffix=".zip") as tmp:
+
+            # Write the zip file
+            for chunk in stream.iter_content(chunk_size=65536):
+                tmp.write(chunk)
+            tmp.seek(0)
+
+            r = self.request(
+                "/api/deposit/depositions/{}/files".format(deposition["id"]),
+                method="POST",
+                data={"filename": "{}.zip".format(self.tale["_id"])},
+                files={"file": tmp},
```

Comment:
Just noting the tmpXXX filename instead of the tale ID.  Looking at the Zenodo docs, I think `filename` should be `name`

```
...
url = 'https://zenodo.org/api/deposit/depositions/1234/files?access_token=ACCESS_TOKEN'
data = {'name': 'myfirstfile.csv'}
files = {'file': open('path/to/local_file.csv', 'rb')}
...
```

Reply:
thanks for nailing it down, fixed in e57f4dc